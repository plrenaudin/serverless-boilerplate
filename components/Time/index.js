import PropTypes from "prop-types";

import "./index.less";

const Time = ({ now = Date.now() }) => (
  <div className="clock">
    The current time,
    <time>{new Date(now).toISOString()}</time>
  </div>
);

Time.propTypes = {
  now: PropTypes.number.isRequired
};

export default Time;
