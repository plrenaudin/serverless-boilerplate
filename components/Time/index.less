@import "../../styles/common.less";

time {
  margin: 0 @spacing;
}
