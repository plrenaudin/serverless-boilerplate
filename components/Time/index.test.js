import Time from ".";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Time now={1548726138665} />).toJSON();
  expect(tree).toMatchSnapshot();
});
