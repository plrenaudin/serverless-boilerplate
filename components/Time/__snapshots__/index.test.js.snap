// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="clock"
>
  The current time,
  <time>
    2019-01-29T01:42:18.665Z
  </time>
</div>
`;
